Tests/UnitTests/Plugins/Geant4/Geant4SurfaceProviderTests.cpp

```diff
@@ -0,0 +1,302 @@
+        double posY = 0;
+        double posZ = nLayer * cm;
+        Acts::Vector3 pos = Acts::Vector3(posX, posY, posZ);
+
+        BOOST_CHECK_EQUAL(
+            sFull.at(nChips * nLayers * nArm + nChips * nLayer + nChip)
+                ->center(gctx),
+            pos);
+      }
+    }
+  }
+
+  // Now check that we can extract only
+  // a subset of the surfaces
+  std::vector<std::string> leftArmNames(names.begin(),
+                                        names.begin() + names.size() / 2);
+
+  auto spLeftArmCfg = Acts::Experimental::Geant4SurfaceProvider<>::Config();
+  spLeftArmCfg.gdmlPath = gdmlPath.string();
+  spLeftArmCfg.surfacePreselector =
+      std::make_shared<Acts::Geant4PhysicalVolumeSelectors::NameSelector>(
+          leftArmNames, true);
+
+  auto spLeftArm =
+      std::make_shared<Acts::Experimental::Geant4SurfaceProvider<>>(
+          spLeftArmCfg);
+
+  auto lbCfg = Acts::Experimental::LayerStructureBuilder::Config();
+  lbCfg.surfacesProvider = spLeftArm;
+
+  auto lbLeftArm =
+      std::make_shared<Acts::Experimental::LayerStructureBuilder>(lbCfg);
+
+  auto [sLeftArm, vLeftArm, suLeftArm, vuLeftArm] = lbLeftArm->construct(gctx);
+
+  BOOST_CHECK_EQUAL(sLeftArm.size(), leftArmNames.size());
+  for (int nLayer = 0; nLayer < nLayers; nLayer++) {
+    for (int nChip = 0; nChip < nChips; nChip++) {
+      double posX = armOffset + nChip * cm;
+      double posY = 0;
+      double posZ = nLayer * cm;
+      Acts::Vector3 pos = Acts::Vector3(posX, posY, posZ);
+
+      BOOST_CHECK_EQUAL(sLeftArm.at(nChips * nLayer + nChip)->center(gctx),
+                        pos);
+    }
+  }
+
+  std::filesystem::remove(gdmlPath);
+}
+
+BOOST_AUTO_TEST_CASE(Geant4SurfaceProviderRanges) {
+  // Construct the world
+  auto [physWorld, names] = ConstructGeant4World();
+
+  // Write GDML file
+  std::filesystem::path gdmlPath = "test.gdml";
+  G4GDMLParser parser;
+  parser.Write(gdmlPath.string(), physWorld);
+
+  // 1D selection -- select only the second row
+  auto sp1DCfg = Acts::Experimental::Geant4SurfaceProvider<1>::Config();
+  sp1DCfg.gdmlPath = gdmlPath.string();
+
+  auto kdt1DOpt = Acts::Experimental::Geant4SurfaceProvider<1>::kdtOptions();
+  kdt1DOpt.range = Acts::RangeXD<1, Acts::ActsScalar>();
+  kdt1DOpt.range[0].set(8, 12);
+  kdt1DOpt.binningValues = {Acts::BinningValue::binZ};
+
+  auto sp1D = std::make_shared<Acts::Experimental::Geant4SurfaceProvider<1>>(
+      sp1DCfg, kdt1DOpt);
+
+  auto lb1DCfg = Acts::Experimental::LayerStructureBuilder::Config();
+  lb1DCfg.surfacesProvider = sp1D;
+
+  auto lb1D =
+      std::make_shared<Acts::Experimental::LayerStructureBuilder>(lb1DCfg);
+
+  auto [s1D, v1D, su1D, vu1D] = lb1D->construct(gctx);
+
+  BOOST_CHECK_EQUAL(s1D.size(), nChips * nArms);
+  for (int nArm = 0; nArm < nArms; nArm++) {
+    for (int nChip = 0; nChip < nChips; nChip++) {
+      int sign = (nArm == 0) ? 1 : -1;
+      double posX = sign * (armOffset + nChip * cm);
+      double posY = 0, posZ = 10;
```

Comment:
```suggestion
      double posY = 0;
      double posZ = 10;
```
Maybe we should separate them to make them clearer.